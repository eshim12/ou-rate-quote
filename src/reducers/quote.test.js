import reducer from './quote'
import {
	GET_QUOTES_REQUESTED,
	GET_QUOTES_SUCCEEDED,
	GET_QUOTES_FAILED
} from '../constants'

describe('quote reducer', () => {
	const initialState = {
		loading: false,
		failed: false,
		error: null,
		data: []
	}

	test('it should return initial state', () => {
		expect(reducer(undefined, {})).toMatchSnapshot()
	})

	test('should handle quotes request', () => {
		expect(reducer(initialState, {type: GET_QUOTES_REQUESTED})).toMatchSnapshot()
	})

	test('should handle successful quotes request', () => {
		expect(reducer(initialState, {type: GET_QUOTES_SUCCEEDED})).toMatchSnapshot()
	})

	test('should handle failed quotes request', () => {
		expect(reducer(initialState, {type: GET_QUOTES_FAILED, payload: {message: "fetch failed"}})).toMatchSnapshot()
	})
})