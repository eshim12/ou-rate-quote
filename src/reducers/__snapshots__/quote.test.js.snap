// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quote reducer it should return initial state 1`] = `Object {}`;

exports[`quote reducer should handle failed quotes request 1`] = `
Object {
  "data": Array [],
  "error": "fetch failed",
  "failed": true,
  "loading": false,
}
`;

exports[`quote reducer should handle quotes request 1`] = `
Object {
  "data": Array [],
  "error": null,
  "failed": false,
  "loading": true,
}
`;

exports[`quote reducer should handle successful quotes request 1`] = `
Object {
  "data": undefined,
  "error": null,
  "failed": false,
  "loading": false,
}
`;
